sessionctx/binloginfo/binloginfo.go

```diff
@@ -112,12 +112,60 @@ var statusListener = func(_ BinlogStatus) error {
 	return nil
 }
 
+// EnableSkipBinlogFlag enables the skipBinlog flag.
+// NOTE: it is used *ONLY* for test.
+func EnableSkipBinlogFlag() {
+	atomic.StoreUint32(&skipBinlog, 1)
+	logutil.BgLogger().Warn("[binloginfo] enable the skipBinlog flag")
+}
+
 // DisableSkipBinlogFlag disable the skipBinlog flag.
 func DisableSkipBinlogFlag() {
 	atomic.StoreUint32(&skipBinlog, 0)
 	logutil.BgLogger().Warn("[binloginfo] disable the skipBinlog flag")
 }
 
+// IsBinlogSkipped gets the skipBinlog flag.
+func IsBinlogSkipped() bool {
+	return atomic.LoadUint32(&skipBinlog) > 0
+}
+
+var committerCounter int32
+
+// WaitBinlogRecover returns when all committing transaction finished.
+func WaitBinlogRecover() {
+	logutil.BgLogger().Warn("[binloginfo] start waiting for binlog recovering")
+	for {
+		if atomic.LoadInt32(&committerCounter) == 0 {
+			break
+		}
+		time.Sleep(time.Second)
+	}
+	logutil.BgLogger().Warn("[binloginfo] binlog recovered")
+}
+
+// CommitterCount returns the number of alive committers.
+// NOTE: it is used *ONLY* for test.
+func CommitterCount() int32 {
+	return atomic.LoadInt32(&committerCounter)
+}
+
+// ResetCommitterCounter is used to reset the committerCounter.
+func ResetCommitterCounter() {
+	atomic.StoreInt32(&committerCounter, 0)
+	logutil.BgLogger().Warn("[binloginfo] committerCounter is reset to 0")
+}
+
+// AddOneCommitter adds one committer to committerCounter.
+func AddOneCommitter() {
```

Comment:
This is used for skipped binlog writer.
Better named to `AddOneSkippedCommiter`